src/Compilers/CSharp/Portable/Compilation/CSharpDiagnosticFilter.cs

```diff
@@ -259,11 +260,18 @@ bool areWarningsGloballyEnabled()
                         if (severity == DiagnosticSeverity.Warning || severity == DiagnosticSeverity.Info)
                         {
                             report = ReportDiagnostic.Suppress;
+                            isSpecified = true;
                         }
                         break;
                 }
             }
 
+            // 5. Global analyzer options
```

Comment:
I believe this should go immediately after `// 3. Compilation level` check. Any rule ID specific configuration should still trump over generic diagnostic option handled in `4. Global options` and be done before pragma related checks. Basically, if global config options suppress a specific diagnostic ID, then it would need a rule ID specific configuration from user to change it.

Reply:
Ah ok, that makes sense I think. I wasn't sure how we wanted to handle the precedence here. 

So, if a NuGet package ups a specific diagnostic to a warning, but the user specifies the global -warn:0 then we would still expect to warn about that particular diagnostic?

Reply:
Ah right yes thinking about it, the user would say -warn:0, but then still have to explicity add a -nowarn for the one the global config upped. That makes sense I think.

Reply:
Yep, any diagnostic ID specific setting should override diagnostic ID agnostic setting. That has been the common design for diagnostic ID configuration from all sources.